packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+  if (resource.abatementString) {
+    const res = isDateWithinDateRange(resource.abatementString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementDateTime) {
+    const res = isDateWithinDateRange(resource.abatementDateTime, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isEncounterWithinDateRange(resource: Encounter, range: DateRange): boolean {
+  if (resource.period) {
+    const res = arePeriodsWithinRange([resource.period], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.location) {
+    const periods = resource.location.map(l => l.period);
+    const res = arePeriodsWithinRange(periods, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isFamilyMemberHistoryWithinDateRange(
+  resource: FamilyMemberHistory,
+  range: DateRange
+): boolean {
+  if (resource.condition) {
+    const periods = resource.condition.map(c => c.onsetPeriod);
+    const arePeriods = arePeriodsWithinRange(periods, range);
+    if (arePeriods !== undefined) return arePeriods;
+
+    const ranges = resource.condition.map(c => c.onsetRange);
+    const areRanges = areRangesWithinRange(ranges, range);
+    if (areRanges !== undefined) return areRanges;
+
+    const dates = resource.condition.flatMap(c => safeDate(c.onsetString) ?? []);
+    const areDates = areDatesWithinRange(dates, range);
+    if (areDates !== undefined) return areDates;
+  }
+  if (resource.date) {
+    const res = areDatesWithinRange([resource.date], range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isDiagnosticReportWithinDateRange(resource: DiagnosticReport, range: DateRange): boolean {
+  const dates = getDatesFromEffectiveDateTimeOrPeriod(resource);
+  const res = areDatesWithinRange(dates, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isDocumentReferenceWithinDateRange(
+  resource: DocumentReference,
+  range: DateRange
+): boolean {
+  const res = isPeriodWithinRange(resource.context?.period, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isConsentWithinDateRange(resource: Consent, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.dateTime, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isCoverageWithinDateRange(resource: Coverage, range: DateRange): boolean {
+  const res = isPeriodWithinRange(resource.period, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isGoalWithinDateRange(resource: Goal, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.startDate, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isImmunizationWithinDateRange(resource: Immunization, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.occurrenceDateTime, range);
```

Comment:
missing resource.recorded + 
 resource.reaction.dateTime

Reply:
good catch.

Do you think its worth to use recorded as a fallback? I think we might end up not adding a resource to the bundle because of the date it was recorded/processed, which might be days/weeks after the date of the actual immunization.

From [the spec](https://www.hl7.org/fhir/R4/immunization.html):

> recorded: When the immunization was first captured in the subject's record

Reply:
Yah I mean I think as a fallback its ok

Reply:
Taken to https://metriport.slack.com/archives/C04DMKE9DME/p1726631358380079?thread_ts=1726522010.103949&cid=C04DMKE9DME